### DISCUSSION TITLE
Create an appendix of all code in a html file

### DISCUSSION THREAD

**Discussion by jhelvy** [quarto-dev/quarto-cli#6650] [Q&A] [answered]
### Description

RMarkdown has a handy way of using the `ref.label` chunk option to grab the code from all of the code chunks in a file and display them all together (see [here](https://bookdown.org/yihui/rmarkdown-cookbook/code-appendix.html)). I've tried implementing this in Quarto but I can't get it to work. Am I missing something? 

I can use `ref.label` to copy the code from a single chunk, but I haven't gotten this to display the code from all chunks.

**Comment by cderv** [comment, marked as answer]
This **knitr** recipe should work in a quarto document. Are you trying to use this in a specific context though ? 
Can you share an example ? 

Here is a working quarto document
````markdown
---
title: "Reproducible Quarto Document"
format: html
execute: 
  echo: false
---

# Content 1

```{r}
plot(mtcars)
```

# Content 2

```{r}
head(mtcars)
```


# Appendix: All code for this report

```{r ref.label=knitr::all_labels()}
#| echo: true
#| eval: false
```
````

For last chunk, I am missing Yaml option and usual knitr syntax, because you need `knitr::all_labels()` to be evaluated
Full YAML syntax would look like 
````markdown
```{r}
#| echo: true
#| eval: false
#| ref-label: !expr knitr::all_labels()
```
````
but 
* You can get R expression in YAML  using `!expr` (specific for **knitr** chunks)
* `ref-label` as yaml chunk option works with latest dev version of **knitr** (otherwise you need to use `ref.label`) 

A note on this: this works for any document that it rendered on the same knitting process. This means that you can do that as the last chunk of a book project document and expect all codes from the book to be included. You need to code around it in that case as each .qmd is knitted in its own independant process. 

Hope it helps

**Reply by jtatria** [reply]
Sorry to revive this old thread, but how would you go about coding around it in the case of book projects? Thanks!

**Comment by jhelvy** [comment]
Yup. that works!

I was doing this:

````
```{r}
#| echo: true
#| eval: false
#| ref.label: knitr::all_labels()
```
````

Didn't realize that it wouldn't be evaluated with that syntax. I prefer the first syntax:

````
```{r ref.label=knitr::all_labels()}
#| echo: true
#| eval: false
```
````

Works like a charm!

**Comment by cl-roberts** [comment]
For anyone who ends up here, I made a lua filter for this use case. See the [`code-appendix`](https://github.com/cl-roberts/.lua-filters/tree/main/code-appendix) repository for the filter or the [show and tell](https://github.com/quarto-dev/quarto-cli/discussions/12498) post that I made.

**Reply by mcanouil** [reply]
There is also "regurgigate" from @coatless

- https://github.com/coatless-quarto/regurgitate
